### ISSUE TITLE
0.3.38 Filling mason jars with methanol from Small Distillery using jars that already had methanol in them results in the game handing me infinite mason jars with methanol

### ISSUE THREAD

**Issue by DragonZephyr** [loiste-interactive/Obenseuer-Issues#1984]
![bug-distilling](https://github.com/loiste-interactive/Obenseuer-Issues/assets/65025239/1d9010a0-67b3-43bb-ad4b-75e73c80d2d9)
As shown in picture more then stack should allow, another issue that plagues me is that items like to stick to my cursor and then i move them to a new slot they just float in the spot they should be but are never moved and thus are stuck in the ether, this happened just before this bug happened in the same slot that is duping the jars but it might be unrelated. 
I have no good repo but with the game handing me jars and not taking them there might be a logic issue in how the stacking of jars with contents is concerned while filling on crafting stations.
Previously i have either used empty jars and then stacked them onto the old ones which is the stack i tried to fill here.

**Comment by DragonZephyr** [comment]
![bug-distilling](https://github.com/loiste-interactive/Obenseuer-Issues/assets/65025239/f79067e8-b19c-4c3b-a22c-35ca413e6b99)
![bug-distilling1](https://github.com/loiste-interactive/Obenseuer-Issues/assets/65025239/1eab37ff-b853-471b-b3df-da9632fa0b08)
The game seems to be dropping empty mason jars outside the crafting station which are either bugged item entities or ghost entities since they do not appear in the inventory when picked up. Performance wise the game handles this quite well despite me trying to break it, there seems to be no check to see if the output of a recipe is higher then it should be as clicking "fill" just increases how much is in the distillery. I have 4 mason jars with 0L Methanol in my inventory which might also be further confusing the game (could be a division error from me dividing jar contents from earlier distillations? I didnt even try to do anything weird)

**Comment by DragonZephyr** [comment]
The game did an admirable job of not putting the mason jars in my walls, i have found 4 wine bottles with ruined mash in them at 1.3 Liters on the ground which i have never placed there. I remember trying to fill wine bottles on the fermenter but it did not let me do it so i do not know if i they are from that or the distillery attempting to spawn them. I have never used wine wine bottles in the mashing tun.
[Moby_Sokolov.zip](https://github.com/loiste-interactive/Obenseuer-Issues/files/15250849/Moby_Sokolov.zip)
2 recent saves of when this was happening included

**Comment by DragonZephyr** [comment]
Experimenting with the very many jars i managed to remove some but this only resulted in normal 8L jars until the slot that was in the distillery integer overflowed forwards eventually allowing me to pick up "infinite out of 8L" methanol jars by moving one into a trash can for no discernible reason it overflowed backwards to NaN out of 8Ls Methanol. Attempting to empty NaN mason jars in the sink resulted in the sink giving me back NaN Mason jars instead of jars that are empty, stacking NaN on Infinite methanol emptied the infinite ones to infinitely empty instead of just making them empty. The game correctly identifies that the jars are empty when i attempt to craft with them but going through a load-zone preserves the bugged state of the mason jars.
Only after i attempted to pull one of my NaN jars out of the chemlab in deek a did the game hang up
[Player.log](https://github.com/loiste-interactive/Obenseuer-Issues/files/15251791/Player.log)
Player.log just after my crash will now attempt to fix my save

**Comment by DragonZephyr** [comment]
Managed to replicate part of the bug, after filling an object with the alcohol i had a full mason jar which i could not fill when i stacked 8 mason jars on my 1 mason jar inside of the small distillery (both had the same quality alcohol in them the single jar was full and the 8 jars were not) i had an empty jar in my inventory and 8 jars in the distillery as expected. However now when i try to fill into this the methanol from the distillery the game produces the Ghost item jars again with every click of fill.

The 1 Mason jar was in the right most slot, potentially tricking the game into thinking the station is full of items when it is not?

Mashing the fill button lets me receive distilling exp but this time i did not get the internal fluid storage to dupe itself. Bug is stable through loading zones however upon loading in the fill button is greyed out despite the mason jar being able to receive, dragging the jar to a slot (even the one it was moved from) updates it thinking it can fill the jar again.
Issue might also be how the game handles 8 mason jars when they need to receive liquid from a crafting recipe. Recipe used was fresh mushroom Kilju from 20L can, bug happened when the can used up the remaining liquid exactly the can nor the liquid inside had no connection with previous instance of the bug but i do not remember if the methanol jar came from unadultered methanol from the previous bug (unlikely though i tried pawning it all of and only kept any bugged jars in a trash can)

[Savegames attached: Moby_Sokolov1.zip](https://github.com/loiste-interactive/Obenseuer-Issues/files/15301218/Moby_Sokolov1.zip)

Autosaves included as i was attempting to see if changing zones had any effect on state.

**Comment by binbows** [comment]
Able to replicate something similar. 
The jars seemingly work as normal. (outside of the duplication)
And jars will spawn on the ground whenever I drag them into my inventory.
https://github.com/user-attachments/assets/2813bf2e-c725-4832-be45-ade74ef1afac

Did encounter other behaviour where the amount of jars would go into the negatives. I wasn't able to document this though, however I did the same as the video shows.

**Comment by Zuntti** [comment]
Same as https://github.com/loiste-interactive/Obenseuer-Issues/issues/2614
